client/src/components/Heart.jsx

```diff
@@ -17,7 +17,7 @@ const Heart = ({ movie, userData, userRating }) => {
       );
       setIsFavorite(favoriteMovie);
     }
-  }, [movie]);
+  }, [movie, userData]);
```

Comment:
I also added this here because of the warning that was complaining about it not being here.. hopefully that is ok?